frontend/src/components/pages/AddProduct.js

```diff
@@ -0,0 +1,232 @@
+import React, { useState } from "react";
+import ProductDataService from "../../client/product";
+import styled from "styled-components";
+
+const AddProduct = () => {
+    const initialProductState = {
+        title: "",
+        description: "",
+        price: "",
+        stock: "",
+        isActive: true,
+        categoryId: ""
+
+    };
+    const [product, setProduct] = useState(initialProductState);
+    const [submitted, setSubmitted] = useState(false);
+
+    const handleInputChange = event => {
+        const { name, value } = event.target;
+        setProduct({...product, [name]: value});
+    };
+
+
+    const saveProduct = () => {
+        var data = {
+            title: product.title,
+            description: product.description,
+            price: product.price,
+            stock: product.stock,
+            isActive: product.isActive,
+            categoryId: product.categoryId
+        };
+
+        ProductDataService.create(data)
+            .then(response => {
+                setProduct({
+                    title: response.data.title,
+                    description: response.data.description,
+                    price: response.data.price,
+                    stock: response.data.stock,
+                    isActive: response.data.isActive,
+                    categoryId: response.data.categoryId
+                });
+                setSubmitted(true);
+                console.log(response.data);
+            })
```

Comment:
Acredito que esse `setProduct` não seja necessário. Afinal, o backend provavelmente vai mandar o mesmo dado que você está enviando, então você está atualizando o estado com os mesmos dados que já estão no estado.

```suggestion
            .then(() => {
                setSubmitted(true);
            })
```